swiftsimio/accelerated.py

```diff
@@ -168,3 +170,271 @@ def index_dataset(handle: Dataset, mask_array: np.array) -> np.array:
     ranges = ranges_from_array(mask_array)
 
     return read_ranges_from_file(handle, ranges, output_size, output_type)
+
+
+################################ ALEXEI: playing around with better read_ranges_from_file implementation #################################
+
+def concatenate_ranges(ranges: np.ndarray) -> np.ndarray:
+    """
+    Returns an array of ranges with consecutive ranges merged if there is no
+    gap between them
+
+
+    Parameters
+    ----------
+    ranges : np.ndarray
+        Array of ranges (see :func:`ranges_from_array`)
+
+    Returns
+    -------
+    np.ndarray
+        two dimensional array of ranges
+    """
+    concatenated = []
+    concatenated.append(ranges[0])
+    
+    for i in range(1,len(ranges)):
```

Comment:
If we are going to have an explicit loop here, please can you @jit the function? We might as well.

Also, this looks like it was not formatted with `black`.